testing/cpm/cpm_generate_pins-nested/verify/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+#=============================================================================
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+cmake_minimum_required(VERSION 3.23.1)
+project(rapids-test-project LANGUAGES CXX)
+
+include(${rapids-cmake-dir}/cpm/init.cmake)
+include(${rapids-cmake-dir}/cpm/package_override.cmake)
+include(${rapids-cmake-dir}/cpm/detail/package_details.cmake)
+
+rapids_cpm_init()
+
+set(projects rmm fmt spdlog CCCL cuco)
```

Comment:
Minor suggestion: to reduce hardcoding (and the risk of future error) you could write the set of packages to load in each subdirectory in the top-level lists file then pass that as a variable when building verify to guarantee they don't get out of sync.

Reply:
Updated the three new tests to do that. Also added in a check to ensure the caller sets the variable so we don't check zero projects.